core/src/main/java/hudson/model/RunMap.java

```diff
@@ -228,9 +228,11 @@ protected R retrieve(File d) throws IOException {
                 }
                 return b;
             } catch (IOException e) {
-                LOGGER.log(Level.WARNING, "could not load " + d, e);
+                throw new IOException();
```

Comment:
Alternately, just leave the original code here untouched, since there was nothing wrong with it.